File: aula5/Binario2.java
public class Binario2 {
	public static void main(String[] args) {
		long q = 0;
		String binario = "";
		boolean erro = false;
		try {
			q = Long.parseLong(args[0]);
		}
		catch (ArrayIndexOutOfBoundsException e) {
			erro = true;
		}
		catch (NumberFormatException e) {
			erro = true;
		}
		if (erro) {
			System.out.println("Digite um número na linha de comando");
			System.exit(-1);
		}
		System.out.println("decimal: " + q);
		while (q >= 2) {
			long resto = q % 2;
			q = q / 2;
			binario = resto + binario;
		}
		binario = q + binario;
		System.out.println("binario: " + binario);		
	}
}

File: aula5/binario2.py
# coding: utf-8
import sys

try:
	q = long(sys.argv[1]);
except (IndexError, ValueError):
	print 'Digite um número na linha de comando'
	sys.exit(-1)
 
print "decimal:", q
binario = ""
while q >= 2:
	resto = q % 2
	q = q / 2
	binario = str(resto) + binario
binario = str(q) + binario
print "binario:", binario
